common/test/acceptance/tests/discussion/test_cohort_management.py

```diff
@@ -824,55 +824,63 @@ def test_always_cohort_inline_topic_enabled(self):
+        And an inline discussion topic with disabled Save button.
         When I click on always_cohort_inline_topics
         Then I see enabled save button
         And I see disabled inline discussion topics
-        When I reload the page
-        Then I see the option enabled
+        When I save the change
+        And I reload the page
+        Then I see the always_cohort_inline_topics option enabled
         """
         self.cohort_discussion_topics_are_visible()
 
-        # enable always inline discussion topics.
+        # enable always inline discussion topics and save the change
         self.cohort_management_page.select_always_inline_discussion()
-
+        self.assertFalse(self.cohort_management_page.is_save_button_disabled(self.inline_key))
         self.assertTrue(self.cohort_management_page.inline_discussion_topics_disabled())
+        self.cohort_management_page.save_discussion_topics(key=self.inline_key)
 
         self.reload_page()
-        self.assertIsNotNone(self.cohort_management_page.always_inline_discussion_selected())
+        self.assertTrue(self.cohort_management_page.always_inline_discussion_selected())
 
     def test_cohort_some_inline_topics_enabled(self):
         """
         Scenario: Select the cohort_some_inline_topics radio button
 
-        Given I have a course with a cohort defined,
-        And a inline discussion topic with disabled Save button.
+        Given I have a course with a cohort defined and always_cohort_inline_topics set to True
+        And an inline discussion topic with disabled Save button.
         When I click on cohort_some_inline_topics
         Then I see enabled save button
         And I see enabled inline discussion topics
-        When I reload the page
-        Then I see the option enabled
+        When I save the change
+        And I reload the page
+        Then I see the cohort_some_inline_topics option enabled
         """
         self.cohort_discussion_topics_are_visible()
+        # By default always inline discussion topics is False. Enable it (and reload the page).
+        self.assertFalse(self.cohort_management_page.always_inline_discussion_selected())
+        self.cohort_management_page.select_always_inline_discussion()
+        self.cohort_management_page.save_discussion_topics(key=self.inline_key)
+        self.reload_page()
+        self.assertFalse(self.cohort_management_page.cohort_some_inline_discussion_selected())
 
         # enable some inline discussion topic radio button.
         self.cohort_management_page.select_cohort_some_inline_discussion()
         # I see that save button is enabled
         self.assertFalse(self.cohort_management_page.is_save_button_disabled(self.inline_key))
         # I see that inline discussion topics are enabled
         self.assertFalse(self.cohort_management_page.inline_discussion_topics_disabled())
+        self.cohort_management_page.save_discussion_topics(key=self.inline_key)
```

Comment:
Note to reviewers: previously this test failed to save the changes. The test didn't fail about line 866 because any empty array is not None (as mentioned above on the helper method).